src/librustc_codegen_llvm/intrinsic.rs

```diff
@@ -724,6 +720,34 @@ impl IntrinsicCallMethods<'tcx> for Builder<'a, 'll, 'tcx> {
         let expect = self.get_intrinsic(&"llvm.expect.i1");
         self.call(expect, &[cond, self.const_bool(expected)], None)
     }
+
+    fn va_start(&mut self, list: &'ll Value) -> &'ll Value {
+        let target = &self.cx.tcx.sess.target.target;
+        let arch = &target.arch;
+        let va_list = match &**arch {
+            _ if target.options.is_like_windows => list,
+            "aarch64" if target.target_os == "ios" => list,
+            "aarch64" | "x86_64" | "powerpc" =>
+                self.load(list, self.tcx().data_layout.pointer_align.abi),
```

Comment:
What's this for exactly? Why are aarch64 (non-iOS) x64, and PowerPC special? Is this documented somewhere? Worth a comment either way.

Reply:
:+1: I'll add a comment. It is due to the architecture specific `va_list` structure used by the three architectures.